python-service/src/report_service.py

```
@@ -0,0 +1,40 @@
+
+    def get_report_by_id(self, report_id):
+        return self.db.execute(
+            "SELECT id, title, author FROM reports WHERE id = ?", (report_id,)
+        ).fetchone()
+
+    def export_report(self, report_id, target_format):
+        cmd = "pandoc " + REPORT_ROOT + "/" + report_id + ".md -o out." + target_format
+        return subprocess.check_output(cmd, shell=True).decode()
+
+    def render_preview(self, report_id):
+        return subprocess.check_output(
```

Ruff (0.15.21)

[error] `subprocess` call: check for execution of untrusted input

(S603)